drivers/volume/volume.go

```diff
@@ -689,3 +698,49 @@ func GetPVCFromObjects(objects []runtime.Unstructured, volumeBackupInfo *storkap
 	}
 	return &pvc, nil
 }
+
+// IsCSIDriverWithoutSnapshotSupport - check whether CSI PV supports snapshot
+// This function will called in csi driver and kdmp volume driver.
+// In csi driver, this api is called to default to kdmp by not owning in CSI driver.
+// In kdmp driver, this api is called to decide whether we need to set the volumesnapclass,
+// to try local snapshot first.
+func IsCSIDriverWithoutSnapshotSupport(pv *v1.PersistentVolume) bool {
+	// check if CSI volume
+	if pv.Spec.CSI != nil {
+		driverName := pv.Spec.CSI.Driver
+		// pure FB csi driver does not support snapshot
+		if driverName == pureCSIProvisioner {
+			if pv.Spec.CSI.VolumeAttributes[pureBackendParam] == pureFileParam {
+				return true
+			}
+		}
+		// vsphere, efs, azure file and google file does not support snapshot.
+		// So defaulting to kdmp by not setting volumesnapshot class.
+		switch driverName {
+		case vSphereCSIProvisioner:
```

Comment:
can we make this as a constant list and if driverName is one among those , then will return true.

Reply:
Fixed.